python/looker_sdk/rtl/api_settings.py

```diff
@@ -49,49 +49,48 @@ def configure(
         instantiate ApiSettings.
 
         ENV variables map like this:
+            LOOKER_API_VERSION -> api_version
             LOOKER_BASE_URL -> base_url
             LOOKER_CLIENT_ID -> client_id
             LOOKER_CLIENT_SECRET -> client_secret
-            LOOKER_EMBED_SECRET -> embed_secret
-
         """
-        cfg_parser = cp.ConfigParser()
-        try:
-            cfg_parser.read_file(open(filename))
-        except FileNotFoundError:
-            config_data: Dict[str, str] = {}
-        else:
-            # If section is not specified, use first section in file
-            section = section or cfg_parser.sections()[0]
-            config_data = dict(cfg_parser[section])
-
-        env_base_url = cast(str, os.getenv("LOOKER_BASE_URL"))
-        if env_base_url:
-            config_data["base_url"] = env_base_url
-
-        env_client_id = cast(str, os.getenv("LOOKER_CLIENT_ID"))
-        if env_client_id:
-            config_data["client_id"] = env_client_id
 
-        env_client_secret = cast(str, os.getenv("LOOKER_CLIENT_SECRET"))
-        if env_client_secret:
-            config_data["client_secret"] = env_client_secret
-
-        env_embed_secret = cast(str, os.getenv("LOOKER_EMBED_SECRET"))
-        if env_embed_secret:
-            config_data["embed_secret"] = env_embed_secret
+        config_data = cls.read_ini(filename, section)
 
         api_version = cast(str, os.getenv("LOOKER_API_VERSION"))
         if api_version:
             config_data["api_version"] = api_version
 
-        # Remove required params from config dictionary if they are empty strings
-        required_params = ["base_url", "client_id", "client_secret"]
-        for key, val in list(config_data.items()):
-            if key in required_params and not val:
-                config_data.pop(key)
+        env_base_url = cast(str, os.getenv("LOOKER_BASE_URL"))
+        if env_base_url:
+            config_data["base_url"] = env_base_url
+
+        required = ["base_url", "client_id", "client_secret"]
```

Comment:
```suggestion
        required = ["base_url", "api_version"]
```

Reply:
`api_version` is not required because the parent class of `ApiSettings`, `transport.TransportSettings` has an `api_version` with a default of `"3.1"`. 

The reason `client_id` and `client_secret` are set to required is so that `ApiSettings` throws an error during the configuration phase if they're not found. But they're not used to instantiate the class since we don't want to store them in memory longer than we need.

Reply:
Gotcha on `api_version`. I didn't check to see how much you mirrored the TS design. 

I'm not sure I'm following how `client_id` and `client_secret` get established here for the required check when they're not properties of the class. I'll look at the updated test code some more to see if I can understand it.